htlcswitch/interceptable_switch.go

```diff
@@ -119,6 +124,14 @@ type FwdResolution struct {
 	// FwdActionSettle.
 	Preimage lntypes.Preimage
 
+	// OutgoingAmountMsat is the amount that is to be used for forwarding if
+	// Action is FwdActionResumeModified.
+	OutgoingAmountMsat lnwire.MilliSatoshi
```

Comment:
May want to consider making these new fields an `fn.Option`, then this way it's a bit clearer that not all of these are always set. THe only thing that's always set is `Key` and `Action`. In the future, we could even consider breaking this down into an interface, then we'd have smaller structs that reflect the type of `FwdAction`.